tests/functional/test_model_completeness.py

```diff
@@ -46,3 +48,48 @@ def test_paginators_and_waiters_are_not_lost_in_new_version(
             raise AssertionError(
                 f"{type_name} must exist for {service_name}: {e}"
             )
+
+
+def _endpoint_rule_set_cases():
+    for service_name in Session().get_available_services():
+        versions = Loader().list_api_versions(service_name, 'service-2')
+        for version in versions:
+            yield service_name, version
+
+
+@pytest.mark.parametrize(
+    "service_name, version",
+    _endpoint_rule_set_cases(),
+)
+def test_all_endpoint_rule_sets_exist(service_name, version):
+    """Tests the existence of endpoint-rule-set.json for each service
+    and verifies that content is present."""
+    loader = Loader()
+    type_name = 'endpoint-rule-set'
+    loader.load_service_model(service_name, type_name, version)
+    full_path = os.path.join(service_name, version, type_name)
+    data = loader.load_data(full_path)
```

Comment:
I'm not sure I understand what is happening here. Isn't this basically doing the same thing twice and could instead be shrunk to 

```suggestion
    data = loader.load_service_model(service_name, type_name, version)
```